modules/irsa/locals.tf

```diff
@@ -17,6 +17,6 @@
  */
 
 locals {
-  eks_oidc_issuer_url   = replace(data.aws_eks_cluster.eks_cluster.identity[0].oidc[0].issuer, "https://", "")
-  eks_oidc_provider_arn = "arn:${data.aws_partition.current.partition}:iam::${data.aws_caller_identity.current.account_id}:oidc-provider/${local.eks_oidc_issuer_url}"
+  eks_oidc_issuer_url   = replace(var.context.aws_eks_cluster.identity[0].oidc[0].issuer, "https://", "")
+  eks_oidc_provider_arn = "arn:${var.context.aws_partition.partition}:iam::${var.context.aws_caller_identity.account_id}:oidc-provider/${local.eks_oidc_issuer_url}"
```

Comment:
Let's bring the actual values from the root module for `eks_oidc_issuer_url` and `eks_oidc_provider_arn`  . See the comment below in root module variables file